modules/dxp/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/provider/AsahSegmentsEntryProvider.java

```diff
@@ -108,39 +106,36 @@ public long[] getSegmentsEntryIds(
 		}
 
 		if (GetterUtil.getBoolean(context.get(Context.SIGNED_IN))) {
-			List<SegmentsEntry> segmentsEntries =
+			List<SegmentsEntry> segmentsEntries = ListUtil.filter(
 				_segmentsEntryLocalService.getSegmentsEntries(
 					groupId, true,
 					SegmentsEntryConstants.SOURCE_ASAH_FARO_BACKEND, className,
-					QueryUtil.ALL_POS, QueryUtil.ALL_POS, null);
-
-			if (segmentsEntries.isEmpty()) {
-				return new long[0];
```

Comment:
@W2jx The original logic has a shortcut when the list obtained from the local service is empty. Please try to keep it.